components/clarinet-deployments/src/diagnostic_digest.rs

```diff
@@ -0,0 +1,110 @@
+use std::collections::HashMap;
+
+use clarity_repl::{
+    clarity::{
+        diagnostic::{Diagnostic, Level},
+        vm::types::QualifiedContractIdentifier,
+    },
+    repl::diagnostic::output_code,
+};
+use colored::*;
+
+use crate::types::DeploymentSpecification;
+
+#[allow(dead_code)]
+pub struct DiagnosticsDigest {
+    pub message: String,
+    pub errors: usize,
+    pub warnings: usize,
+    pub contracts_checked: usize,
+    full_success: usize,
+    total: usize,
+}
+
+impl DiagnosticsDigest {
+    pub fn new(
+        contracts_diags: &HashMap<QualifiedContractIdentifier, Vec<Diagnostic>>,
+        deployment: &DeploymentSpecification,
+    ) -> DiagnosticsDigest {
+        let mut full_success = 0;
+        let mut warnings = 0;
+        let mut errors = 0;
+        let mut contracts_checked = 0;
+        let mut outputs = vec![];
+        let total = deployment.contracts.len();
+
+        for (contract_id, diags) in contracts_diags.into_iter() {
+            let (source, contract_location) = match deployment.contracts.get(&contract_id) {
+                Some(entry) => {
+                    contracts_checked += 1;
+                    entry
+                }
+                None => {
+                    // `deployment.contracts` only includes contracts from the project, requirements should be ignored
+                    continue;
+                }
+            };
+            if diags.is_empty() {
+                full_success += 1;
+                continue;
+            }
+
+            let lines = source.lines();
+            let formatted_lines: Vec<String> = lines.map(|l| l.to_string()).collect();
+
+            for diagnostic in diags {
+                match diagnostic.level {
+                    Level::Error => {
+                        errors += 1;
+                        outputs.push(format!("{} {}", "error:".red().bold(), diagnostic.message));
```

Comment:
Using `colored::red` helpers instead of hirosystems' kit for better portability (wasm)